source/src_pdiag/diag_cusolver.cuh

```diff
@@ -0,0 +1,51 @@
+#include <complex>
+
+#include "nvToolsExt.h"
+#include <cuda_runtime.h>
+#include <cusolverDn.h>
+
+
+class Diag_Cusolver_gvd{
+
+    cusolverDnHandle_t cusolverH;
+    cusolverStatus_t cusolver_status;
+    cudaError_t cudaStat1;
```

Comment:
There's no need to storing error status. I suggest capturing CUDA exceptions by macro, see https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L595 .

Reply:
This style is from https://docs.nvidia.com/cuda/archive/9.1/cusolver/index.html#sygvd-example1. If needed for abacus, I will change to the macro style.

Reply:
> This style is from https://docs.nvidia.com/cuda/archive/9.1/cusolver/index.html#sygvd-example1. If needed for abacus, I will change to the macro style.

This style derives from cuda 9.1 (might be outdated, I think), and I suggest following the latest convention(for `cuSOLVER`, see https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L205).

Reply:
> > This style is from https://docs.nvidia.com/cuda/archive/9.1/cusolver/index.html#sygvd-example1. If needed for abacus, I will change to the macro style.
> 
> This style derives from cuda 9.1 (might be outdated, I think), and I suggest following the latest convention(for `cuSOLVER`, see https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L205).

The link is pasted just for convenience and the codes are the same with those in the cuda v11.1 cusolver pdf document. But I think macro style may be better and it has been accepted in the commit ready to push.